etcdserver/server.go

```diff
@@ -938,7 +941,12 @@ func (s *EtcdServer) run() {
 					s.compactor.Resume()
 				}
 			}
-
+			if newLeader {
+				select {
+				case s.leaderChanged <- struct{}{}:
```

Comment:
if we want to do a broadcast style notification, we should close this channel not just sending one item.

Reply:
What do you mean?  seems I do not need to broadcast the leader change, it just consumes the channel in `linearizableReadLoop`. There are two possibilities of being blocked in the loop, one is no read request before the leader changed, and another is that there is a read request coming during the leader changed process. Although there are two places to consume this channel, it only consumes once. cc @xiang90

Reply:
The chan is named as `leaderChanged`. By reading the variable name, people will assume they will get notification when there is a leader change by using this chan. However, when this chan gets used by multiple routines, this chan only deliver one notification. There will be a problem.

I would rather make this chan a broadcast chan.